src/WebJobs.Extensions.DurableTask/ContextImplementations/DurableActivityContext.cs

```diff
@@ -55,13 +55,30 @@ internal JToken GetInputAsJson()
         {
             if (this.serializedInput != null && this.parsedJsonInput == null)
             {
-                JArray array = JArray.Parse(this.serializedInput);
-                if (array?.Count != 1)
+                if (this.messageDataConverter.JsonSettings.TypeNameHandling == TypeNameHandling.All)
```

Comment:
I think the fundamental problem here is not just with TypeNameHandling.All, but that we don't use the JsonSettings on the message data converter to deserialize here.

Reply:
You're right, I could make this the only path instead. For some reason initially I just figured that because in other cases it deserializes to JArray I should keep it like that because its simpler. But that doesn't make sense, I can just choose to deserialize to an object[] every time.